hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py

```diff
@@ -113,40 +93,40 @@ def __init__(self,
                  binance_perpetual_api_secret: str = None,
                  trading_pairs: Optional[List[str]] = None,
                  trading_required: bool = True,
-                 **domain):
-        self._testnet = True if len(domain) > 0 else False
-        ExchangeBase.__init__(self)
-        PerpetualTrading.__init__(self)
-        BinanceTime.get_instance().start()
+                 domain: str = "binance_perpetual",
+                 ):
         self._api_key = binance_perpetual_api_key
         self._api_secret = binance_perpetual_api_secret
+        self._trading_pairs = trading_pairs
         self._trading_required = trading_required
-        # self._account_balances = {}
-        # self._account_available_balances = {}
+        self._throttler = AsyncThrottler(CONSTANTS.RATE_LIMITS)
+        self._domain = domain
 
-        self._base_url = PERPETUAL_BASE_URL if self._testnet is False else TESTNET_BASE_URL
-        self._stream_url = DIFF_STREAM_URL if self._testnet is False else TESTNET_STREAM_URL
-        self._user_stream_tracker = BinancePerpetualUserStreamTracker(base_url=self._base_url, stream_url=self._stream_url, api_key=self._api_key)
-        self._order_book_tracker = BinancePerpetualOrderBookTracker(trading_pairs=trading_pairs, **domain)
+        ExchangeBase.__init__(self)
+        PerpetualTrading.__init__(self)
+        BinanceTime.get_instance().start()
```

Comment:
So in the `ConnectorBase` class, there is a call to the `name` property. Since we are using `self._domain` as the return value of the `name` property, declaring `self._domain` after the initialization of the `ExchangeBase` would result in a `AttributeError`. That being said, I am in favour of having `self._domain` being declared before `ExchangeBase.__init__(self)` and the rest being declared after.